src/Symfony/Component/Console/Command/Command.php

```diff
@@ -79,36 +90,7 @@ public static function getDefaultDescription(): ?string
      */
     public function __construct(?string $name = null)
     {
-        $this->definition = new InputDefinition();
-
-        if (null === $name && null !== $name = static::getDefaultName()) {
```

Comment:
The methods `getDefaultName()` and `getDefaultDescription()` were not final, they can be overridden in child classes. You moved their logic have into the new `init()` method, but you have to call them is they are defined in a child class, and trigger a deprecation message.

You can use `ReflectionMethod::getDeclaringClass()`.

```php
if ((new ReflectionMethod($this, 'getDefaultName'))->getDeclaringClass() !== Command::class`) {
    trigger_deprecation('symfony/console', '7.3', 'The static method "%s()" is deprecated and will be removed in Symfony 8.0, extract the command name from the "%s" attribute instead.', __METHOD__, AsCommand::class);
    $name = static::getDefaultName();
}
```